src/checks/endsWithBytes.ts

```diff
@@ -2,16 +2,14 @@ import { open, stat } from 'fs/promises';
 
 import { CheckFunction } from '../types';
 
-export function endsWithByteArray(
-  endByteArray: ArrayLike<number>,
-): CheckFunction {
-  if (!Array.isArray(endByteArray)) {
-    throw new TypeError('endByteArray should be an array');
+export function endsWithBytes(endBytes: ArrayLike<number>): CheckFunction {
+  if (!Array.isArray(endBytes)) {
```

Comment:
This is going to fail if `endBytes` is an Uint8Array. can you assert that the length is at least 1 ? Actually we should also check that in the endsWithStr function